### DISCUSSION TITLE
How can i add file icons ? sudo make 0_NERD=1  doesn't work

### DISCUSSION THREAD

**Discussion by miraculusik** [jarun/nnn#1579] [Q&A] [answered]
OS: MacOS
Nerd Font: ComicCodeLigatures Nerd Font
Terminal: Alacritty

As I mentioned in the title, when I patch to add the icons it doesn't work. I'm using nerd font in my terminal and i am not getting any error.
```
sudo make 0_NERD=1
output: 
cc  -std=c11 -Wall -Wextra -Wshadow -O3   -o nnn  src/nnn.c -lreadline -lncurses -lpthread
```
this code create `nnn` but it's not showing file icons.

**Comment by N-R-K** [comment]
1. What option are you running `nnn` with? (Make sure you're not using `-C`)
2. What's the value of `$TERM` ? Does using `TERM=xterm-256color nnn` work ?

**Reply by miraculusik** [reply]
1. i am running in the nnn folder with just typing`./nnn` (I'm doing this just to see if it's working)
2.  echo $TERM = xterm-256color

**Reply by N-R-K** [reply]
> i am running in the nnn folder with just typing./nnn (I'm doing this just to see if it's working)

`nnn` can still be configured from the enviornment variables. Post the output of `env | grep -i 'nnn'`

**Reply by miraculusik** [reply]
> > i am running in the nnn folder with just typing./nnn (I'm doing this just to see if it's working)
> 
> `nnn` can still be configured from the enviornment variables. Post the output of `env | grep -i 'nnn'`
```
PWD=$HOME/nnn
NNN_FCOLORS=0404040000000600010F0F02
NNN_PLUGINS=i:imgview
```

**Reply by N-R-K** [reply]
I'm out of ideas then. Perhaps your system `locale` is not set up properly (see https://github.com/jarun/nnn/issues/1518#issuecomment-1312694219).

**Comment by jarun** [comment]
Try `O_EMOJI=1`. Should work out of the box in most cases.

**Reply by miraculusik** [reply]
It doesn't work.

**Reply by jarun** [reply]
What's your locale output?

**Reply by jarun** [reply]
Also, share a screenshot of what you see with `O_EMOJI=1`.

**Reply by jarun** [reply]
Q3. What is the output of `make O_EMOJI=1`?

For me it's:

```
$ make O_EMOJI=1
cc -DEMOJI -DICONS_INCLUDE=\"icons-generated-emoji.h\" -std=c11 -Wall -Wextra -Wshadow -O3 -D_DEFAULT_SOURCE -D_XOPEN_SOURCE=600  -DICONS_GENERATE -o src/icons-hash-gen src/icons-hash.c
./src/icons-hash-gen > src/icons-generated-emoji.h
cc -DEMOJI -DICONS_INCLUDE=\"icons-generated-emoji.h\" -std=c11 -Wall -Wextra -Wshadow -O3 -D_DEFAULT_SOURCE -D_XOPEN_SOURCE=600  -o nnn  src/nnn.c -lreadline -lncursesw -ltinfo -lpthread
```

**Reply by miraculusik** [reply]
> What's your locale output?

```
LANG=""
LC_COLLATE="C"
LC_CTYPE="UTF-8"
LC_MESSAGES="C"
LC_MONETARY="C"
LC_NUMERIC="C"
LC_TIME="C"
LC_ALL=
```

**Reply by miraculusik** [reply]
> Q3. What is the output of `make O_EMOJI=1`?
> 
> For me it's:
> 
> ```
> $ make O_EMOJI=1
> cc -DEMOJI -DICONS_INCLUDE=\"icons-generated-emoji.h\" -std=c11 -Wall -Wextra -Wshadow -O3 -D_DEFAULT_SOURCE -D_XOPEN_SOURCE=600  -DICONS_GENERATE -o src/icons-hash-gen src/icons-hash.c
> ./src/icons-hash-gen > src/icons-generated-emoji.h
> cc -DEMOJI -DICONS_INCLUDE=\"icons-generated-emoji.h\" -std=c11 -Wall -Wextra -Wshadow -O3 -D_DEFAULT_SOURCE -D_XOPEN_SOURCE=600  -o nnn  src/nnn.c -lreadline -lncursesw -ltinfo -lpthread
> ```

```
cc  -std=c11 -Wall -Wextra -Wshadow -O3   -o nnn  src/nnn.c -lreadline -lncurses -lpthread
```

**Reply by jarun** [reply]
See the difference in the output of the `make` command? For you the icons are not even compiling. Are you on master?

**Reply by miraculusik** [reply]
I think the problem is here but I can't quite figure out what I'm missing

**Reply by miraculusik** [reply]
> See the difference in the output of the `make` command? For you the icons are not even compiling. Are you on master?

yes

**Reply by jarun** [reply]
Run the following 2 commands manually and see if that works:

```
cc -DEMOJI -DICONS_INCLUDE=\"icons-generated-emoji.h\" -std=c11 -Wall -Wextra -Wshadow -O3 -D_DEFAULT_SOURCE -D_XOPEN_SOURCE=600  -DICONS_GENERATE -o src/icons-hash-gen src/icons-hash.c
./src/icons-hash-gen > src/icons-generated-emoji.h
cc -DEMOJI -DICONS_INCLUDE=\"icons-generated-emoji.h\" -std=c11 -Wall -Wextra -Wshadow -O3 -D_DEFAULT_SOURCE -D_XOPEN_SOURCE=600  -o nnn  src/nnn.c -lreadline -lncursesw -ltinfo -lpthread
```

**Comment by N-R-K** [comment]
> sudo make 0_NERD=1

1. It's O (english alphabet O) not 0 (zero). Try again with `O_NERD=1` or `O_EMOJI=1`

2. LANG is not set in your locale output, which might be playing a part in this. My locale output is this:

```console
$ locale
LANG=C.UTF8
LC_CTYPE="C.UTF8"
LC_NUMERIC="C.UTF8"
LC_TIME="C.UTF8"
LC_COLLATE="C.UTF8"
LC_MONETARY="C.UTF8"
LC_MESSAGES="C.UTF8"
LC_PAPER="C.UTF8"
LC_NAME="C.UTF8"
LC_ADDRESS="C.UTF8"
LC_TELEPHONE="C.UTF8"
LC_MEASUREMENT="C.UTF8"
LC_IDENTIFICATION="C.UTF8"
LC_ALL=
```

**Reply by miraculusik** [reply]
It's WORKKK YES!. Thank you so much :)

**Comment by jarun** [comment, marked as answer]
OK. Here's the problem: you used `0_NERD=1` (numeric zero) instead of `O_NERD=1` (capital O).

**Reply by miraculusik** [reply]
It's work Thank you so much :) I have typing issue

**Reply by jarun** [reply]
Please mark as answered so I can have a new badge. ;)

**Reply by miraculusik** [reply]
> Please mark as answered so I can have a new badge. ;)

Congrats 🎉🎉🎉
